internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -780,31 +794,39 @@ func (c *Coordinator) handleCoordinatorDone(ctx context.Context, varsErrCh, runt
 				timeouts = append(timeouts, "no response from varsWatcher component")
 			}
 			c.logger.Debugf("timeout while waiting for other components to shut down: %v", timeouts)
-			break
+			doneWait = true
 		case err := <-runtimeErrCh:
 			runtimeErr = err
+			returnedRuntime = true
 		case err := <-configErrCh:
+			returnedConfig = true
 			configErr = err
 		case err := <-varsErrCh:
+			returnedVars = true
 			varsErr = err
 		}
-		if runtimeErr != nil && configErr != nil && varsErr != nil {
-			timeoutWait.Stop()
-			break
+		if returnedRuntime && returnedConfig && returnedVars {
+			c.logger.Debugf("all error channels have returned in handleCoordinatorDone")
+			doneWait = true
 		}
+
 	}
 	// try not to lose any errors
 	var combinedErr error
 	if runtimeErr != nil && !errors.Is(runtimeErr, context.Canceled) {
+		c.logger.Debugf("runtime component error is non-nil during shutdown: %s", runtimeErr)
```

Comment:
```suggestion
		c.logger.Debugf("runtime component error during shutdown: %s", runtimeErr)
```

You don't need the `is non-nil` because we don't log nil errors. This applies to all of the lines below.